File: EventsDetailFragment/app/src/main/java/com/example/eventsdetailfragment/Fragments/EventDetailsFragment.java
package com.example.eventsdetailfragment.Fragments;

import android.content.Context;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.EditText;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import com.example.eventsdetailfragment.FragmentCommunication;
import com.example.eventsdetailfragment.R;

public class EventDetailsFragment extends Fragment {

    private Button btnNext;
    private EditText etTitle, etDescription;
    private FragmentCommunication communication;

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        return inflater.inflate(R.layout.fragment_event_details, container, false);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        initViews(view);
    }

    private void initViews(View view) {
        btnNext = view.findViewById(R.id.btnNext);
        etTitle = view.findViewById(R.id.etTitle);
        etDescription = view.findViewById(R.id.etDescription);

        btnNext.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Bundle bundle = new Bundle();
                String title = etTitle.getText().toString();
                String Description = etDescription.getText().toString();
                bundle.putString("title", title);
                bundle.putString("description", Description);
                if (communication != null) {
                    communication.launchSecondFrag(bundle);
                }
            }
        });
    }

    @Override
    public void onAttach(@NonNull Context context) {
        super.onAttach(context);
        communication = (FragmentCommunication) context;
    }
}

File: Kotlin Projects/EventsNavFragments/app/src/main/java/com/example/events_navfragments/EventDetailsFragment.kt
package com.example.events_navfragments

import android.os.Bundle
import android.view.View
import androidx.fragment.app.Fragment
import androidx.navigation.NavController
import androidx.navigation.Navigation
import kotlinx.android.synthetic.main.fragment_event_details.*


class EventDetailsFragment : Fragment(R.layout.fragment_event_details) {

    private lateinit var navController: NavController


    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        navController = Navigation.findNavController(view)

        btnNext.setOnClickListener {
            val bundle = Bundle()
            val eventName = etEventName.text.toString()
            val eventDesc = etEventDecs.text.toString()

            bundle.putString("eventName", eventName)
            bundle.putString("eventDesc", eventDesc)
            navController.navigate(R.id.action_eventDetailsFragment_to_timeAndDateFragment, bundle)

        }
    }

}